### PR TITLE
fixed array.split_axis when combined labels are not sorted by the first part then second part (closes #364)

### LINKED ISSUE #364: split_axis broken when combined labels are not sorted by the first part then second part

### PR DESCRIPTION
**Pull request by gdementen**

### CODE REVIEW

File: larray/core/array.py [hunk 1]
```diff
@@ -6591,7 +6591,7 @@ def split_axis(self, axis, sep='_', names=None, regex=None):
          a0   0   1   2
          a1   3   4   5
         """
-        return self.reshape(self.axes.split_axis(axis, sep, names, regex))
+        return self.sort_axis(axis).reshape(self.axes.split_axis(axis, sep, names, regex))
```
Comment (alixdamman) [inline larray/core/array.py]: why not handle this as an option which is True by default. Users may unsorted axis after split?

Comment (gdementen) [reply larray/core/array.py]: because it mangles the data and I do not see that as an interesting option ;-)

Comment (alixdamman) [reply larray/core/array.py]: OK

**Comment by gdementen** [comment]
my fix is broken...

**Comment by gdementen** [comment]
Here is the proper fix. The remaining problem concerns #376.

File: larray/core/axis.py [hunk 2]
```diff
@@ -274,6 +274,64 @@ def extend(self, labels):
...
 
+    def split(self, sep='_', names=None, regex=None, return_labels=False):
+        """Split axis and returns a list of Axis.
+
+        Parameters
+        ----------
+        sep : str, optional
+            Delimiter to use for splitting. Defaults to '_'.
+
+            When `regex` is provided, the delimiter is only used on `names` if given as one string or on axis name if
```
Comment (alixdamman) [inline larray/core/axis.py]: why a blank line? To force line return?

Comment (gdementen) [reply larray/core/axis.py]: no idea. I think it was that way where I copy-paste that block from :) (or I introduced it by mistake)

File: larray/core/axis.py [hunk 3]
```diff
@@ -274,6 +274,64 @@ def extend(self, labels):
...
+            `names` is None.
+        names : str or list of str, optional
+            Names of resulting axes. Defaults to None.
+        regex : str, optional
+            Use regex instead of delimiter to split labels. Defaults to None.
+        labels : bool, optional
+            Whether or not split labels must be returned (as a tuple of tuples). These labels are suitable for indexing
+            via array.points[labels]. Defaults to False.
```
Comment (alixdamman) [inline larray/core/axis.py]: Do you want to open a new issue to add that option to LArray.split_axes?

Comment (gdementen) [reply larray/core/axis.py]: LArray.split_axes? I don't see a use case. In AxisCollection.split_axes, that **might** make the code of LArray.split_axes nicer/smaller, but I doubt the array.points[] = value with multiple labels (for the several axes) will work. I haven't thought much about that though.

**Comment by alixdamman** [comment]
except for the extra blank line in docstring of `Axis.split`, LGTM

**Pull request merged**